system/include/compat/avxintrin.h

```diff
@@ -4,20 +4,18 @@
  * University of Illinois/NCSA Open Source License.  Both these licenses can be
  * found in the LICENSE file.
  */
+
+#ifndef __emscripten_immintrin_h__
```

Comment:
Let's use a different macro that gets undefined after all the relevant includes in immintrin.h. Otherwise, we won't emit this error if the user does something like this:

```
#include <immintrin.h>
#include <avxintrin.h>
```

Reply:
Yes, in this case, it won't emit error message, but I think it's suitable. 

Actually, include avxintrin.h directly  will miss prerequisites and cause compile error. Include immintrin.h can avoid this error. 
immintrin.h includes the necessary prerequisites and choose the *intrin.h corresponding to options(-mavx, -mavx2...).
so in the following code, if -mavx option is specified, #include <avxintrin.h> does nothing due to header guards, it's valid code, the behavior is consistent with LLVM.

```
#include <immintrin.h>
#include <avxintrin.h>
```